src/typings/react-native.d.ts

```diff
@@ -267,6 +267,13 @@ declare module 'react-native' {
         // iOS
         onAccessibilityTapIOS?: Function;
         shouldRasterizeIOS? : boolean;
+
+        // Windows
+        onKeyDown?: Function;
```

Comment:
Yikes, I don't like the idea of adding these props to all components. These should be added selectively only to the components where they make sense. Some of these are even duplicates of existing props, for example the onHoverStart/End for RX.Button.

Any place that we're adding props for keyboard or mouse we should be asking ourselves why we don't also add support for these same methods on the web (and eventually MacOS). They should be equally applicable.

Also, any callbacks should be strongly typed, not generic "Function" types.

Reply:
You mentioned RX.Button, so you may think this change is in Typings.ts. It's actually in react-native.d.ts.
onKeyDown handles bubbled events raised by children.
onMouse* are supported by a range of RN for W views. They map to RX.ViewPropShared.onMouse* properties, similar to what the web counterpart does with <div>.onMouse* events.

---
In reply to: [159786010](https://github.com/Microsoft/reactxp/pull/404#discussion_r159786010) [](ancestors = 159786010)

Reply:
Ah, my bad. Yes, for some reason, I thought this was Typings.ts. Never mind.